### ISSUE TITLE
Failure in runing Nunaliit v. 2.2.3 on Mac OS X

### ISSUE THREAD

**Issue by pulsifer** [GCRC/nunaliit#347]
Hi,

I am attempting to run the distributionv version of Nunaliit v. 2.2.3 on Max OS X version 10.9.5

As set of errors are presented (see below)

It appears that this may be related to the version of Java being used?  I am running:

java version "1.8.0_45"
Java(TM) SE Runtime Environment (build 1.8.0_45-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.45-b02, mixed mode)

Thoughts on the root of the problem?

Errors

Exception in thread "main" java.lang.UnsupportedClassVersionError: javax/servlet/ServletException : Unsupported major.minor version 51.0
    at java.lang.ClassLoader.defineClass1(Native Method)
    at java.lang.ClassLoader.defineClassCond(ClassLoader.java:637)
    at java.lang.ClassLoader.defineClass(ClassLoader.java:621)
    at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:141)
    at java.net.URLClassLoader.defineClass(URLClassLoader.java:283)
    at java.net.URLClassLoader.access$000(URLClassLoader.java:58)
    at java.net.URLClassLoader$1.run(URLClassLoader.java:197)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.net.URLClassLoader.findClass(URLClassLoader.java:190)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:306)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:301)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:247)
    at ca.carleton.gcrc.couch.command.Main.getCommands(Main.java:31)
    at ca.carleton.gcrc.couch.command.Main.execute(Main.java:120)
    at ca.carleton.gcrc.couch.command.Main.main(Main.java:55)

**Comment by jpfiset** [comment]
Yes, it looks like your platform installed a newer version of Java. Is it possible to install another version (1.7) of Java on this platform? If not, we might need to recompile a version of 2.2.3 for Java 1.8

**Comment by pulsifer** [comment]
Let me try to install 1.7.  Will
Let you know but probably not until Monday.

**Comment by pulsifer** [comment]
I did manage to uninstall Java 1.8.  I am now running the following:

java version "1.6.0_65"
Java(TM) SE Runtime Environment (build 1.6.0_65-b14-462-11M4609)
Java HotSpot(TM) 64-Bit Server VM (build 20.65-b04-462, mixed mode)

This presents the same type of error as version 1.8 (see below), so I suspect that Version 1.7 is what I need.  However, this brings up a larger question which is whether using an older (possibly less secure) version of Java is ok.  My organization indicates that the latest version of Java should be used.  I can experiment to see if I can run both 1.7 and 1.8 and point to 1.7 for Nunaliit?

I did try to build Nunaliit assuming that this would avoid this Java version issue, however the build is failing which I will submit as separate ticket after checking for similar issues here and if I cannot resolve the build problems.

Is it problematic to maintain versions of Nunaliit that are compiled for different versions of Java?  Compiling a version on Java 1.8 would resolve the issue in the short-term, however may not be a sustainable solution.

**Comment by rodolpheg** [comment]
Any luck solving this issue yet?
Thanks!

**Comment by ahayes** [comment]
@rodolpheg Are you running into this issue?

We don't build or test the server-side of Nunaliit on OS X but would welcome any pull requests necessary to make it work. We currently build the Nunaliit distribution bundle on Java 7 on Ubuntu.

**Comment by rodolpheg** [comment]
I am running into a similar error on MacOS 10.12. Here it is:

MacBook-Pro$ nunaliit create
Exception in thread "main" java.lang.UnsupportedClassVersionError: ca/carleton/gcrc/couch/command/Main : Unsupported major.minor version 51.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClassCond(ClassLoader.java:637)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:621)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:141)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:283)
	at java.net.URLClassLoader.access$000(URLClassLoader.java:58)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:197)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:190)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:306)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:301)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:247)

I don't have a Linux Machine at hand, unfortunately. I'll try and make a pull request, thanks for the suggestion!

**Comment by ahayes** [comment]
Hmm.. Sorry about that. You may want to try cloning the repository using git and building from source using the developer docs. If you find a way to make it work, please let us know!

Virtual Box is free and Ubuntu 16.04 server is free and relatively small. So if you want to give it a shot without solving that problem, it might be faster.

**Comment by rodolpheg** [comment]
Building from source worked perfectly with the developper docs. I simply had to install couchdb (brew install couchdb) and Java (brew cask install java) before hand. Thanks!

**Comment by boikle** [comment]
This still an issue?

**Comment by ahayes** [comment]
No longer relevant. Closing.
